### DISCUSSION TITLE
random number

### DISCUSSION THREAD

**Discussion by Jaap1957** [nrc-cnrc/EGSnrc#1095] [Programming]
Hi,
I am working with an e- sensor. The e- beam (electron microscope) deposition is nicely calculated. Eenrgy is converted to e-h pairs at ~3.6eV per pair. I want to follow the paors.

For that i need a random direction generator. It seems 

EGS_Vector x = inside_shape->getRandomPoint(rndm);
( line 273,  HEN_HOUSE\egs++\egs_geometry_tester.cpp )

would suit the purpose.

How do I, can I, use this function in a top CPP application. f..e tutor7pp ?

Thanks !

**Comment by Jaap1957** [comment]
This line inserted in tutor7pp.cpp. compiles with egsnrc_GUI:

`EGS_Float zz = rndm->getGaussian();`

I just guessed rndm-> would associate to functions in egs_rndm.h
with simalr lines for x and y one can form a random directed vctor:

```                 EGS_Float zz = 0;
                    while (zz==0) {
                        EGS_Float zz = rndm->getGaussian();
                    }
                    EGS_Float xx = rndm->getGaussian();
                    EGS_Float yy = rndm->getGaussian();
```
this construction prevents a (0,0,0) outcome.

**Reply by rtownson** [reply]
It sounds like you have solved it. FYI, if you are looking for an isotropic directional distribution, check out the `getPositionDirection` function in `HEN_HOUSE/egs++/egs_point_source`.

**Reply by ftessier** [reply]
@Jaap1957 you ought to check that the vector length is non-zero (and then normalize). The way you have it, all point on the equator (z=0) will be overlooked!
